### ISSUE TITLE
Face and head loss are NAN, when training PyramidBox. Online Urgent!

### ISSUE THREAD

**Issue by YueXiNPU** [PaddlePaddle/models#1282]
Hello everyone and @qingqing01. I am training _PyramidBox (ECCV 2018)_ by Baidu with 3 GPUs of GTX 1080Ti, memory 11G. Training command is _python -u train.py --batch_size=3_. I try batch_size=4/5/6, but it does not work for me.
Nan is shown as below:
![default](https://user-images.githubusercontent.com/34571270/45916935-62bf6780-beb1-11e8-99ec-4461bf3900a7.png)
GPU information is shown as below:
![default](https://user-images.githubusercontent.com/34571270/45916944-b5991f00-beb1-11e8-9064-2e9db0166597.png)
Anyone can help me to solve this problem?

**Comment by YueXiNPU** [comment]
@qingqing01, would be very appreciated if you can give us any feedback or update.

**Comment by YueXiNPU** [comment]
There are lots of things I have seen make a model diverge.
1, Too high of a learning rate(LR). The default LR is 0.001. I tried LR = 0.5e-3/0.1e-3/0.5e-4/0.1e-4. But it does not work for us.
2,  I will check the input data and labels.  Make sure you are not introducing the nan. Also make sure all of the target values are valid.  Make sure the data is properly normalized. You probably want to have the pixels in the range [-1, 1] and not [0, 255].

Best
Yue

**Comment by qingqing01** [comment]
Do you use the pre-trainde vgg model?

**Comment by YueXiNPU** [comment]
@qingqing01 Sure, As for this command _python -u train.py --batch_size=3_ , I use the default parameter --pretrained_model=./vgg_ilsvrc_16_fc_reduced/.

**Comment by neverland0621** [comment]
@qingqing01 同问+1.
命令语句是
export CUDA_VISIBLE_DEVICES=0,1,2
python -u train.py --batch_size=3 --pretrained_model=vgg_ilsvrc_16_fc_reduced
显卡配置是3张1080ti（11G）
在pass 0，batch_size =750时出现face_loss和head_loss都等于nan。处理时间为0.61.
麻烦解答，谢谢 ^_^

**Comment by YueXiNPU** [comment]
Would be very appreciated if you could give some suggestions to us. Because we have dedicated ourselves to the training stage. @qingqing01

**Comment by qingqing01** [comment]
Generally speaking, can reduce lr.  But I need to do some experiments. We nerver trained this model to used batch size 1 per gpu.

**Comment by qingqing01** [comment]
Update some experiments. I use the latest develop code and batch size is 3, use 3 GPUs:

```
-----------  Configuration Arguments -----------
batch_size: 3
data_dir: /paddle/data/WIDERFACE/
epoc_num: 160
learning_rate: 0.001
mean_BGR: 104., 117., 123.
model_save_dir: output
parallel: True
pretrained_model: ./vgg_ilsvrc_16_fc_reduced/
resize_h: 640
resize_w: 640
use_gpu: True
use_pyramidbox: True
with_mem_opt: True
------------------------------------------------
```

```
Pass 0, batch 1320, face loss 4.96722173691, head loss 4.77904748917, time 0.809649944305
Pass 0, batch 1330, face loss 3.80087780952, head loss 3.2623269558, time 0.571543931961
Pass 0, batch 1340, face loss 5.90743589401, head loss 4.51720571518, time 0.762489080429
Pass 0, batch 1350, face loss 3.84677004814, head loss 2.95647501945, time 0.629281997681
```

Still no NaN.

**Comment by takecareofbigboss** [comment]
Hi, maybe it is owe to lacking of gt faces in the image '0_Parade_Parade_0_275.jpg', you can have a try to delete this file in the gt file.   Otherwise, you can enlarge the batchsize for each gpu.

**Comment by takecareofbigboss** [comment]
@YueXiNPU @neverland0621

**Comment by YueXiNPU** [comment]
@qingqing01 @takecareofbigboss Many thanks for your comments.

**Comment by neverland0621** [comment]
Thank you very much for your help. As you mentioned,
I didn't find the groundtruth of image "0_Parade_Parade_0_275.jpg ", so I deleted the image in file "WIDER_val" .  Unfortunately, it didn't work for the problem of loss  "NAN".
I doubt that the data we train has other similar problems like this .  What's worse,
When I training the model,my machine often shut down by itself .
Could u give me advice to solve the problem ?
